File: 输入输出练习/3.cpp
/*
 * @Description: 
 * @Author: Mr. Lee
 * @Date: 2021-07-26 22:27:18
 * @LastEditTime: 2021-07-26 22:29:57
 * @LastEditors: Mr. Lee
 */

#include <iostream>

using namespace std;

int Add(int a, int b)
{
    return a + b;
}

int main()
{
    int A, B;
    cin >> A >> B;
    while (A != 0 || B != 0)
    {
        cout << Add(A, B) << endl;
        cin >> A >> B;
    }
    return 0;
}

File: 秋招笔试/拼多多笔试/3.py
'''
Description: 
Author: Mr. Lee
Date: 2021-07-25 19:49:22
LastEditTime: 2021-07-26 17:06:12
LastEditors: Mr. Lee
'''


def func(A, B, C, Q):
    if B == 0 or C == 0:
        return 0
    elif (Q == A) or ((Q - A) % B == 0) or (((Q - A) // B) % C == 0):
        return 1

    return 0


if __name__ == '__main__':
    import sys

    n = int(sys.stdin.readline().strip())
    for _ in range(n):
        a, b, c, q = list(map(int, sys.stdin.readline().strip().split()))
        print(func(a, b, c, q))
